packages/core/contracts/UnirepSocial.sol

```diff
@@ -291,6 +316,7 @@ contract UnirepSocial is zkSNARKHelper {
             downvoteValue,
             publicSignals[4] // min rep
         );
+        usedProofNullifier[proofNullifier] = true;
```

Comment:
can you set this earlier in the function? we want to avoid re-entry vulnerabilities after external calls.